src/test/java/com/commercetools/service/impl/OrderServiceImplIntegrationTest.java

```diff
@@ -0,0 +1,86 @@
+package com.commercetools.service.impl;
+
+import com.commercetools.Application;
+import com.commercetools.service.OrderService;
+import com.commercetools.testUtil.ctpUtil.CtpResourcesUtil;
+import com.commercetools.testUtil.customTestConfigs.CartsCleanupConfiguration;
+import com.commercetools.testUtil.customTestConfigs.OrdersCleanupConfiguration;
+import com.commercetools.testUtil.customTestConfigs.PaymentsCleanupConfiguration;
+import com.commercetools.testUtil.customTestConfigs.TaxSetupConfig;
+import io.sphere.sdk.carts.Cart;
+import io.sphere.sdk.carts.CartDraft;
+import io.sphere.sdk.carts.commands.CartCreateCommand;
+import io.sphere.sdk.carts.commands.CartUpdateCommand;
+import io.sphere.sdk.carts.commands.updateactions.AddPayment;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.orders.Order;
+import io.sphere.sdk.orders.OrderFromCartDraft;
+import io.sphere.sdk.orders.commands.OrderFromCartCreateCommand;
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.payments.PaymentDraftBuilder;
+import io.sphere.sdk.payments.PaymentDraftDsl;
+import io.sphere.sdk.payments.commands.PaymentCreateCommand;
+import io.sphere.sdk.taxcategories.TaxCategory;
+import io.sphere.sdk.taxcategories.queries.TaxCategoryQuery;
+import org.javamoney.moneta.Money;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.Optional;
+
+import static com.commercetools.testUtil.CompletionStageUtil.executeBlocking;
+import static com.commercetools.testUtil.customTestConfigs.TaxSetupConfig.TAX_CATEGORY_NAME;
+import static io.sphere.sdk.models.DefaultCurrencyUnits.EUR;
+import static org.assertj.core.api.Assertions.assertThat;
+
+// TODO: move to separate integration test
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = Application.class)
+@Import(value = {
+        OrdersCleanupConfiguration.class,
+        CartsCleanupConfiguration.class,
+        PaymentsCleanupConfiguration.class,
+        TaxSetupConfig.class
+})
+public class OrderServiceImplIntegrationTest {
+
+    @Autowired
+    private SphereClient sphereClient;
+
+    @Autowired
+    private OrderService orderService;
+
+    @Test
+    public void createOrderManuallyAndGetByPaymentId() {
+        PaymentDraftDsl paymentDraft = PaymentDraftBuilder.of(Money.of(22.33, EUR))
+                .build();
+        Payment ctPayment = executeBlocking(sphereClient.execute(PaymentCreateCommand.of(paymentDraft)));
+
+        createOrderWithPayment(ctPayment);
+
+        Optional<Order> orderOpt = executeBlocking(orderService.getByPaymentId(ctPayment.getId()));
+        assertThat(orderOpt.isPresent()).isTrue();
```

Comment:
Yes, but we don't have this version of assertJ, since we're using spring-boot-starter-test and it has only 2.6.0.: https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-test/1.5.4.RELEASE
I can upgrade to newest boot-start-test, WDYT? https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-test/2.0.0.M2